src/core/lib/iomgr/resource_quota.cc

```diff
@@ -430,39 +451,53 @@ static bool rq_reclaim(grpc_resource_quota* resource_quota, bool destructive) {
  * ru_slice: a slice implementation that is backed by a grpc_resource_user
  */
 
-typedef struct {
+static void ru_slice_ref(void* p);
+static void ru_slice_unref(void* p);
+static const grpc_slice_refcount_vtable ru_slice_vtable = {
+    ru_slice_ref, ru_slice_unref, grpc_slice_default_eq_impl,
+    grpc_slice_default_hash_impl};
+
+namespace {
+struct ru_slice_refcount {
```

Comment:
Can this just use our existing `RefCounted` implementation?

I would understand if there are some problems here, since the slice refcounting API is C style.  But it's worth trying to avoid the duplicate refcounting code if possible.

Reply:
Sorry I missed this one I think. There are many of such structs in the code. The problem is many of these structs have to be castable to their base, so if we wanted to add a refcounted base class we would need multiple inheritance, which doesn't feel right.

https://github.com/grpc/grpc/blob/618a3f561d4a93f263cca23abad086ed8f4d5e86/src/core/ext/transport/chttp2/transport/internal.h#L222

Reply:
I'd like feedback from @nicolasnoble on this.  IIRC, he had a proposal for how we might be able to address this issue.